core/src/wsv.rs

```diff
@@ -33,53 +28,54 @@ pub type NewBlockNotificationSender = tokio::sync::watch::Sender<()>;
 /// Receiver type of the new block notification channel
 pub type NewBlockNotificationReceiver = tokio::sync::watch::Receiver<()>;
 
-/// World trait for mocking
-pub trait WorldTrait:
-    Deref<Target = World> + DerefMut + Send + Sync + 'static + Debug + Default + Sized + Clone
-{
-    /// Creates a [`World`] with these [`Domain`]s and trusted [`PeerId`]s.
-    fn with(
-        domains: impl IntoIterator<Item = Domain>,
-        trusted_peers_ids: impl IntoIterator<Item = PeerId>,
-    ) -> Self;
-}
-
 /// The global entity consisting of `domains`, `triggers` and etc.
 /// For example registration of domain, will have this as an ISI target.
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Default, Clone, Getters)]
+#[cfg_attr(feature = "mock", allow(dead_code))]
+#[getset(get = "pub")]
 pub struct World {
     /// Iroha parameters.
-    pub parameters: Vec<Parameter>,
+    parameters: Vec<Parameter>,
```

Comment:
If you insist on using `getset` here you will get `fn parameters(&self) -> &Vec<Parameter>` where I think the declaration should be `fn parameters(&self) -> &[Parameter]`

Reply:
I'd prefer this be a separate `impl` block. This is idiomatically a slice, which should be treated as such.